### ISSUE TITLE
Enable discussions

### ISSUE THREAD

**Issue by ramonemiliani93** [tweag/python-monorepo-example#13]
Hi there! 👋

I wanted to express my gratitude, Guillaume and Clément, for the insightful blog posts you've shared. I found them truly fascinating! I'm reaching out here to inquire about the possibility of enabling discussions, and to kindly ask if you're open to addressing some queries regarding the design decisions you've made and your approach to handling certain aspects.

I've been attempting to follow your mono repo approach and have been particularly intrigued by your insights into docker-compose files for development within a project. To illustrate, suppose I'm actively working on both "Library A" and "Project A." I've been exploring ways to establish a Docker environment that facilitates seamless updates to both the library and the project, while ensuring that any changes I make are accurately reflected within the containers. In the context of "Project A" functioning as a Flask application or something akin to it, I'm yet to discover a method that allows me to achieve this, especially when utilizing editable installs.

Warm regards,

Ramón

**Comment by smelc** [comment]
Hey @ramonemiliani93, thanks for you kind words :heart_eyes_cat: and happy to discuss the setup we proposed (which is by no way a rule-them-all setup!) and the differences with your setup.

If I were to reflect updates to the repository's content in a running container, I suppose I would use Docker's _volume mapping_ and read the mapped files in the container. Is it something  you tried?

**Comment by ramonemiliani93** [comment]
Yes, that's usually the recommended approach for these cases. However, a specific issue arises with the setup when you're simultaneously working on a project and a library. Since the library dependency is defined in the requirements.txt file, you're faced with a choice: you can either copy the library, install it, and the changes won't be reflected; or you can mount it using a volume mapper, installing it through the Dockerfile command, with the caveat that the requirements will be installed each time.

Maybe there is a solution for this I haven't thought about yet 😅 Thank's again for the blog post series 🙌 It is indeed a very nice setup (there are always pros and cons).

**Comment by GuillaumeDesforges** [comment]
Hi @ramonemiliani93, thanks for the good question!

If you use poetry to package your project and library (or setuptools < 64), its installation will only add a `.pth` file to the site-packages, which has the path to the folder with the library.

So in theory (I have not tried) you should be able to override the directory in the container that has the lib ( the one where you copied your library) by mounting from the directory in the host where you have your lib source. Then your changes on the host should be reflected inside the container.

See https://stackoverflow.com/questions/47664107/docker-mount-to-folder-overriding-content

If you try this approach, I'm interested to know if it works well for you :+1:
